File: 3600-find-the-k-th-character-in-string-game-i/3600-find-the-k-th-character-in-string-game-i.py
class Solution:
    def kthCharacter(self, k: int) -> str:
        ans = "a"
        ref = "abcdefghijklmnopqrstuvwxyz"

        while(len(ans) < k ):
            cur = ans
            for i in ans:
                cur = cur + ref[((ord(i) - ord('a')) + 1) % 26]
            ans = cur
        
        return ans[k - 1]

File: 3600-find-the-k-th-character-in-string-game-i/3600-find-the-k-th-character-in-string-game-i.cpp
class Solution {
public:
    char kthCharacter(int k) {
        string ans = "a", ref = "abcdefghijklmnopqrstuvwxyz";
        while(ans.length() < k){
            string cur = ans;
            for(char ch : ans){
                cur += ref[((ch-'a') + 1) % 26];
            }
            ans = cur;
        }
        return ans[k - 1];
    }
};